doc/releases/changelog-dev.md

```diff
@@ -384,29 +405,94 @@
   DeviceArray(0., dtype=float32)
   ```
 
+<h4>Operations and batching</h4>
+
+* A new function `dot` has been added to compute the dot product between a vector and a list of operators.
+  [(#3586)](https://github.com/PennyLaneAI/pennylane/pull/3586)
+
+  ```pycon
+  >>> coeffs = np.array([1.1, 2.2])
+  >>> ops = [qml.PauliX(0), qml.PauliY(0)]
+  >>> qml.dot(coeffs, ops)
+  (1.1*(PauliX(wires=[0]))) + (2.2*(PauliY(wires=[0])))
+  >>> qml.dot(coeffs, ops, pauli=True)
+  1.1 * X(0) + 2.2 * Y(0)
+  ```
+
+* `qml.evolve` returns the evolution of an `Operator` or a `ParametrizedHamiltonian`.
```

Comment:
For me, the dream would be that the changelog allows you to click on the `qml.evolve` and it will take you to the relevant place in the docs - which removes some of the uncertainty around whether we should have `qml.` or something else.

This is already possible in the main PL docs, and the changelog now lives in the PL docs - but the `.md` version of the changelog on GH may not look great.